src/iocp/mod.rs

```diff
@@ -45,20 +45,30 @@ use std::marker::PhantomPinned;
 use std::os::windows::io::{AsRawHandle, RawHandle, RawSocket};
 use std::pin::Pin;
 use std::sync::atomic::{AtomicBool, Ordering};
-use std::sync::{Arc, Mutex, MutexGuard, RwLock};
+use std::sync::{Arc, Mutex, MutexGuard, RwLock, Weak};
 use std::time::{Duration, Instant};
 
 #[cfg(not(polling_no_io_safety))]
 use std::os::windows::io::{AsHandle, BorrowedHandle};
 
+/// Macro to lock and ignore lock poisoning.
+macro_rules! lock {
+    ($lock_result:expr) => {{
+        ($lock_result).unwrap_or_else(|e| e.into_inner())
```

Comment:
Rust has hygienic macros, so the parentheses around `$lock_result` are unnecessary.